pkg/syfs/syfs.go

```diff
@@ -61,6 +62,15 @@ func configDir(dir string) string {
 	}
 
 	homedir := os.Getenv("HOME")
+	needUpdate := false
+	if homedir != "" && !fs.IsWritable(homedir) {
+		// apptainer process can not write the homedir, should ignore the manual value
+		sylog.Warningf("Environment variable `HOME` is set to a non-writable path:%s, will reset it", homedir)
+		// reset the homedir based on user's info and current working dir
+		homedir = ""
```

Comment:
But setting homedir to empty here is going to be immediately followed up just below by looking up the current user id from the password and making it non-empty again.  In the case of the original issue, the real user's home directory is non-readable, not just `$HOME`, so this will not fix that case.  Does it truly work to have an empty homedir?  Maybe it would work if it changed to "/" instead of being empty, and check done after the empty homedir code below.  I know that ssh for example will switch to "/" if the user's home directory isn't accessible.

Reply:
thanks! I forgot that the home directory is non-readable here, so setting the home dir value based on the current user's info will ultimately fail. So maybe we can set it to `/nonexistent` dir as shown in the `/etc/passwd` for user `nobody`? this actually has the similar behavior of setting it to `/`.